packages/contracts-bedrock/contracts/L1/OptimismPortal.sol

```diff
@@ -162,32 +162,39 @@ contract OptimismPortal is Initializable, ResourceMetering, Semver {
     }
 
     /**
-     * @notice Determine if an L2 Output is finalized.
+     * @notice Determine if an L2 Output is finalized for a given block number.
      *
      * @param _l2BlockNumber The number of the L2 block.
      */
+    function isOutputFinalized(uint256 _l2BlockNumber) public view returns (bool) {
+        L2OutputOracle.OutputProposal memory proposal;
 
-    function isOutputFinalized(uint256 _l2BlockNumber) external view returns (bool) {
-        L2OutputOracle.OutputProposal memory proposal = L2_ORACLE.getL2Output(_l2BlockNumber);
+        uint256 interval = L2_ORACLE.SUBMISSION_INTERVAL();
+        uint256 startingBlockNumber = L2_ORACLE.STARTING_BLOCK_NUMBER();
+        if (startingBlockNumber > _l2BlockNumber) {
+            return false;
+        }
+
+        // Find the distance between _l2BlockNumber, and the checkpoint block before it.
+        uint256 offset = (_l2BlockNumber - startingBlockNumber) % interval;
+
+        // If the offset is zero, then the _l2BlockNumber should be checkpointed.
+        // Otherwise, we'll look up the next block that will be checkpointed.
+        uint256 lookupBlockNumber = offset == 0
+            ? _l2BlockNumber
+            : _l2BlockNumber + (interval - offset);
 
-        if (proposal.outputRoot == bytes32(uint256(0))) {
-            uint256 interval = L2_ORACLE.SUBMISSION_INTERVAL();
-            uint256 startingBlockNumber = L2_ORACLE.STARTING_BLOCK_NUMBER();
-
-            // Prevent underflow
-            if (startingBlockNumber > _l2BlockNumber) {
-                return false;
-            }
-
-            // Find the distance between the _l2BlockNumber, and the checkpoint block before it.
-            uint256 offset = (_l2BlockNumber - startingBlockNumber) % interval;
-            // Look up the checkpoint block after it.
-            proposal = L2_ORACLE.getL2Output(_l2BlockNumber + (interval - offset));
-            // False if that block is not yet proposed.
-            if (proposal.outputRoot == bytes32(uint256(0))) {
-                return false;
-            }
+        try L2_ORACLE.getL2Output(lookupBlockNumber) returns (
```

Comment:
I don't think the try/catch is necessary, I think it's fine if this reverts instead of returning false.